### PR TITLE
Adds documentation and news post for secure container-to-container...

### LINKED ISSUE #1969: Add container-to-container encryption example to docs
- [x] Update cloud.gov docs as needed
- [x] Create new blog post outlining the new functionality
- [x] Create sample app demonstrating how it works for possible use in upcoming tech talk

### PR DESCRIPTION
**Pull request by dandersonsw**
…networking

## Changes proposed in this pull request:
- Adds docs and news post for secure container-to-container networking
- Fixes #1969 
-

:sunglasses:[PREVIEW URL](https://cg-88d42ca6-59d7-47e0-9500-4dd9251360b9.app.cloud.gov/preview/cloud-gov/cg-site/BRANCH_NAME)

## Security Considerations
None

### CODE REVIEW

File: _posts/2022-04-06-encrypted-container-to-container-networking.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+date: February 9, 2022
+title: "New Feature: Encrypted Container-to-Container Networking"
+excerpt: A new update to Container-to-Container networking allows developers to enable encryption of traffic between applications using SSL/TLS.
+---
+
+The cloud.gov platform now offers developers the ability to encrypt container-to-container traffic using SSL/TLS, providing more security for your applications internal communications. This feature is also another step towards zero trust security.
+
+## About Container-to-Container Networking
+
+Container-to-container networking allows applications to communicate with each other over an internal network known as an [overlay network](https://docs.cloudfoundry.org/concepts/understand-cf-networking.html#overlay-network). Traffic sent over this overlay network is not allowed to leave said network, and the network itself is not externally routable. This ensures that nothing outside of the network can see the traffic being passed between applications.
```
Comment (mogul) [inline _posts/2022-04-06-encrypted-container-to-container-networking.md]:
```suggestion
Container-to-container networking allows applications to communicate with each other over an internal network known as an [overlay network](https://docs.cloudfoundry.org/concepts/understand-cf-networking.html#overlay-network). Traffic sent over this overlay network is not allowed to leave it, and the network itself is not externally routable. This ensures that nothing outside of the network can see the traffic being passed between applications.
```

Comment (mogul) [reply _posts/2022-04-06-encrypted-container-to-container-networking.md]: (plain language)

File: _posts/2022-04-06-encrypted-container-to-container-networking.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+layout: post
+date: February 9, 2022
```
Comment (mogul) [inline _posts/2022-04-06-encrypted-container-to-container-networking.md]:
```suggestion
date: April 6, 2022
```

File: _posts/2022-04-06-encrypted-container-to-container-networking.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+In terms of the zero trust security model this addresses the assumption that just because something is on the internal network it should be implicitly trusted. Instead, traffic is encrypted so that only the specified applications involved can see the data.
+
+Cloud Foundry, and by extension cloud.gov, has designated port `61443` to be used for encrypted container-to-container traffic. Any traffic sent between applications using this port will automatically be encrypted. This port can be specified using the `cf add-network-policy` command.
+
+When setting up encryption the destination application itself does not need to be changed, and the source application only needs to be modified to send traffic to the correct port. For a more in-depth guide check out the [container-to-container networking]({{ site.baseurl }}{% link _docs/management/container-to-container.md %}) documentation. You can also check out the official Cloud Foundry documentation on [Securing Container-to-Container Traffic](https://docs.cloudfoundry.org/concepts/understand-cf-networking.html#securing-traffic).
```
Comment (mogul) [inline _posts/2022-04-06-encrypted-container-to-container-networking.md]:
```suggestion
When setting up encryption the destination application itself does not need to be changed, and the source application only needs to be modified to send traffic to the correct port. If your application is TLS-aware, you still have the option to use a different port and terminate the encryption yourself, using a certificate provided in the application container.

For a more in-depth guide check out the [container-to-container networking]({{ site.baseurl }}{% link _docs/management/container-to-container.md %}) documentation. You can also check out the official Cloud Foundry documentation on [Securing Container-to-Container Traffic](https://docs.cloudfoundry.org/concepts/understand-cf-networking.html#securing-traffic).
```

File: _posts/2022-04-06-encrypted-container-to-container-networking.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+## Securing Container-to-Container Networking
+
+Container-to-container networking on its own works to keep traffic between applications from being accessed by the outside world. However, this does not prevent other applications or users on the internal network from accessing the data being transferred between applications. This is where secure container-to-container networking comes in. By encrypting the internal traffic between applications nothing else will be able to read the information being sent.
+
+In terms of the zero trust security model this addresses the assumption that just because something is on the internal network it should be implicitly trusted. Instead, traffic is encrypted so that only the specified applications involved can see the data.
+
+Cloud Foundry, and by extension cloud.gov, has designated port `61443` to be used for encrypted container-to-container traffic. Any traffic sent between applications using this port will automatically be encrypted. This port can be specified using the `cf add-network-policy` command.
```
Comment (mogul) [inline _posts/2022-04-06-encrypted-container-to-container-networking.md]:
```suggestion
Cloud Foundry, and by extension cloud.gov, has designated port `61443` to be used for encrypted container-to-container traffic. Any traffic sent between applications using this port will be encrypted automatically. This port can be specified using the `cf add-network-policy` command.
```

File: _docs/management/container-to-container.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+To set up and allow traffic between your applications:
+
+1. Create an internal route to your application(s)
+```
+cf map-route APP apps.internal --hostname APP
+```
+2. Add a network policy to allow traffic from one application to another
+```
+cf add-network-policy SOURCE_APP DESTINATION_APP -s DESTINATION_SPACE_NAME -o DESTINATION_ORG_NAME --protocol (tcp | udp) --port RANGE
+```
```
Comment (bengerman13) [inline _docs/management/container-to-container.md]: indent the backticks and code line so they'll be part of the ordered list and not restart the numbering
~~~suggestion
1. Create an internal route to your application(s)
   ```
   cf map-route APP apps.internal --hostname APP
   ```
2. Add a network policy to allow traffic from one application to another
   ```
   cf add-network-policy SOURCE_APP DESTINATION_APP -s DESTINATION_SPACE_NAME -o DESTINATION_ORG_NAME --protocol (tcp | udp) --port RANGE
   ```
~~~

File: _docs/management/container-to-container.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+## Configuring secure container-to-container networking
+
+Two options are available for configuring encryption of traffic between your applications using SSL/TLS: `automatic` and `manual`. With the automatic option the platform handles everything, from provisioning certificates to ensuring TLS termination at the destination, and specifically uses port `61443`. The manual option involves configuring your application with its own TLS termination, and can be configured to use any available port.
+
+To set up automatic encryption:
+
+1. Create a network policy for your destination application and set the port to 61443. All traffic sent to this port will use SSL/TLS
+```
+cf add-network-policy SOURCE_APP DESTINATION_APP -s DESTINATION_SPACE_NAME -o DESTINATION_ORG_NAME --protocol tcp --port 61443
+```
```
Comment (bengerman13) [inline _docs/management/container-to-container.md]:
~~~suggestion
   ```
   cf add-network-policy SOURCE_APP DESTINATION_APP -s DESTINATION_SPACE_NAME -o DESTINATION_ORG_NAME --protocol tcp --port 61443
   ```
~~~

File: _docs/management/container-to-container.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+cf map-route APP apps.internal --hostname APP
+```
+2. Add a network policy to allow traffic from one application to another
+```
+cf add-network-policy SOURCE_APP DESTINATION_APP -s DESTINATION_SPACE_NAME -o DESTINATION_ORG_NAME --protocol (tcp | udp) --port RANGE
+```
+
+## Configuring secure container-to-container networking
+
+Two options are available for configuring encryption of traffic between your applications using SSL/TLS: `automatic` and `manual`. With the automatic option the platform handles everything, from provisioning certificates to ensuring TLS termination at the destination, and specifically uses port `61443`. The manual option involves configuring your application with its own TLS termination, and can be configured to use any available port.
```
Comment (bengerman13) [inline _docs/management/container-to-container.md]: I think we can make this a little less wordy:

```suggestion
There are two main ways to enable TLS between your application: 
- Automatically - With this option the platform handles everything, from provisioning certificates to enforcing TLS termination at the destination (steps described below)
- Manually - This option is necessary if you want to listen on multiple or non-configurable ports. With this option, you are responsible for configuring your application with its own TLS termination and managing TLS certificates. The steps to enable this option depend on the application(s) you're running, and are out of scope for this documentation.
```

**Review by mogul — APPROVED** [review]

**Pull request merged**